### PR TITLE
feat(install): Separate namespaced and descoped rbacs

### PR DESCRIPTION
**Pull request by gansheer**
Ref #3165

* Update config rbacs files to have separate full namespaced and global (descoped) version. 
* Adapt the CLI and kustomize installation

**Release Note**
```release-note
feat(install): Separate namespaced and descoped rbacs 
```

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by squakez — APPROVED** [review]

File: pkg/install/operator.go [hunk 1]
```diff
@@ -200,44 +199,6 @@ func OperatorOrCollect(ctx context.Context, cmd *cobra.Command, c client.Client,
...
-			}
-
-			if rb, ok := o.(*rbacv1.RoleBinding); ok {
-				if strings.HasPrefix(rb.Name, "camel-k-operator") {
-					rb.Subjects[0].Namespace = cfg.Namespace
-
-					o = &rbacv1.ClusterRoleBinding{
-						ObjectMeta: metav1.ObjectMeta{
-							Namespace: cfg.Namespace,
-							Name:      fmt.Sprintf("%s-%s", rb.Name, cfg.Namespace),
```
Comment (gansheer) [inline pkg/install/operator.go:227]: @lburgazzoli @phantomjinx Do you know why the global cluster role bindings name are replaces with the namespace suffix ?

**Comment by gansheer** [comment]
@oscerd @christophd @squakez Could you trigger the e2e tests please :pray:

File: pkg/install/operator.go (hunk 1 above)
Comment (claudio4j) [reply pkg/install/operator.go:227]: It comes from #2683, to allow multiple global operators in multiple namespaces, to create a CRB and associate to a SA which was installed in the global operator namespace.

**Comment by gansheer** [comment]
@squakez  can you launch the tests again please 🙏 ?

**Comment by squakez** [comment]
@gansheer is this good to merge? how is this affecting the installation procedures (if it does affect in any manner)?

**Comment by gansheer** [comment]
It is ready for merge. From my tests, on vanilla kubernetes and Openshift, it is transparent on existing installation methods CLI and Kustomize (I didn't test helm as helm uses its own files).

File: config/rbac/kustomization.yaml [hunk 2]
```diff
@@ -23,23 +23,22 @@ kind: Kustomization
...
-- operator-role-binding.yaml
 - operator-cluster-role-custom-resource-definitions.yaml
 - operator-cluster-role-binding-custom-resource-definitions.yaml
 - operator-cluster-role-addressable-resolver.yaml
 - operator-cluster-role-binding-addressable-resolver.yaml
 - operator-cluster-role-local-registry.yaml
+- operator-role-binding-local-registry.yaml
+
+
+transformers:
```
Comment (squakez) [inline config/rbac/kustomization.yaml:34]: Is this one something which would require user's attention by any chance? I mean, does it deserve some kind of documentation?

Comment (gansheer) [reply config/rbac/kustomization.yaml:34]: @squakez I added the documentation :book:

**Review by squakez — APPROVED** [review]
Good for now, but we need to provide some simplification in the future.

**Pull request merged**